src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -415,25 +389,25 @@
   [config catalog]
   {:pre  [(catalog? catalog)]
    :post [(catalog? %)]}
-  (let [context {"node" (get catalog ["certname"])}]
+  (let [context {"node" (get catalog "certname")}]
     (-> catalog
-        (update-in ["resources"]        anonymize-resources context config)
-        (update-in ["edges"]            anonymize-edges context config)
-        (update-in ["certname"]         anonymize-leaf :node context config)
-        (update-in ["transaction_uuid"] anonymize-leaf :transaction_uuid context config)
-        (update-in ["environment"] anonymize-leaf :environment context config))))
+        (update "resources" anonymize-catalog-resources context config)
+        (update "edges" anonymize-edges context config)
+        (update "certname" anonymize-leaf :node context config)
+        (update "transaction_uuid" anonymize-leaf :transaction_uuid context config)
+        (update "environment" anonymize-leaf :environment context config))))
 
 (pls/defn-validated anonymize-report :- report-schema-str
   "Anonymize a report"
   [config report :- report-schema-str]
   (let [context {"node" (get report "certname")}]
     (-> report
-        (update-in ["certname"]         anonymize-leaf :node context config)
-        (update-in ["resource_events"]  anonymize-resource-events context config)
-        (update-in ["metrics"] anonymize-metrics context config)
-        (update-in ["logs"] anonymize-logs context config)
-        (update-in ["transaction_uuid"] anonymize-leaf :transaction_uuid context config)
-        (update-in ["environment"] anonymize-leaf :environment context config))))
+        (update "certname" anonymize-leaf :node context config)
+        (update "resources" anonymize-report-resources context config)
```

Comment:
It doesn't actually matter for metrics/logs/resources, because (map foo nil) will return (), which are valid values. But I'll change it for consistency.